extensions/guacamole-auth-quickconnect/src/main/java/org/apache/guacamole/auth/quickconnect/QuickConnectDirectory.java

```diff
@@ -106,6 +114,12 @@ public String create(GuacamoleConfiguration config) throws GuacamoleException {
         // Generate a name for the configuration.
         String name = QCParser.getName(config);
 
+        // Purge configuration parameters that are either not allowed or are
+        // explicitly denied.
+        QCParser.purgeConfigParameters(config,
+                confService.getAllowedParameters(),
+                confService.getDeniedParameters());
```

Comment:
Manually purging parameters after the `GuacamoleConfiguration` has been created makes me uncomfortable. If this is ever accidentally omitted, then a security issue appears. Can this be done at the time that the `GuacamoleConfiguration` is built from the URL?

For example, if `QCParser` were concrete, with a constructor requiring a set of allowed and denied parameter, that would ensure that all parsing of URLs has to explicitly take the allow/deny bits into account, and any usage that (unsafely) _ignores_ those bits would need to be pretty purposefully written to ignore them.

Reply:
Sure, makes sense. I've got the changes almost done - just need to clean up the unit tests.

Reply:
Okay, things have been re-worked as suggested.